src/type_inference/stage1.coffee

```diff
@@ -200,7 +200,11 @@ type_generalize = require "../type_generalize"
     
     when "Var_decl"
       if root.assign_value
-        root.assign_value.type = ti.type_spread_left root.assign_value.type, root.type, ctx
+        if root.assign_value.val_list
```

Comment:
It's better to check type first with .constructor.name == "Struct_init", not to check specific field presense